sdmetrics/reports/single_table/_properties/data_validity.py

```diff
@@ -0,0 +1,124 @@
+        'id': KeyUniqueness,
+    }
+
+    def _generate_details(self, real_data, synthetic_data, metadata, progress_bar=None):
+        """Generate the _details dataframe for the data validity property.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data
+            synthetic_data (pandas.DataFrame):
+                The synthetic data
+            metadata (dict):
+                The metadata of the table
+            progress_bar (tqdm.tqdm or None):
+                The progress bar to use. Defaults to None.
+        """
+        column_names, metric_names, scores = [], [], []
+        error_messages = []
+        for column_name in metadata['columns']:
+            sdtype = metadata['columns'][column_name]['sdtype']
+            primary_key_match = column_name == metadata.get('primary_key')
+            alternate_key_match = column_name in metadata.get('alternate_keys', [])
+            is_unique = primary_key_match or alternate_key_match
+
+            try:
+                if sdtype not in self._sdtype_to_metric and not is_unique:
+                    continue
+
+                metric = self._sdtype_to_metric.get(sdtype, KeyUniqueness)
+                column_score = metric.compute(
+                    real_data[column_name], synthetic_data[column_name]
+                )
+                error_message = None
+
+            except Exception as e:
+                column_score = np.nan
+                error_message = f'{type(e).__name__}: {e}'
+            finally:
+                if progress_bar:
+                    progress_bar.update()
+
+            column_names.append(column_name)
+            metric_names.append(metric.__name__)
+            scores.append(column_score)
+            error_messages.append(error_message)
+
+        result = pd.DataFrame({
+            'Column': column_names,
+            'Metric': metric_names,
+            'Score': scores,
+            'Error': error_messages,
+        })
+
+        if result['Error'].isna().all():
+            result = result.drop('Error', axis=1)
+
+        return result
+
+    def get_visualization(self):
+        """Create a plot to show the data validity scores.
+
+        Returns:
+            plotly.graph_objects._figure.Figure
+        """
+        average_score = round(self._compute_average(), 2)
+
+        fig = px.bar(
+            data_frame=self.details,
+            x='Column',
+            y='Score',
+            title=f'Data Diagnostic: Data Validity (Average Score={average_score})',
+            category_orders={'group': list(self.details['Column'])},
+            color='Metric',
+            color_discrete_map={
+                'BoundaryAdherence': PlotConfig.DATACEBO_DARK,
+                'CategoryAdherence': PlotConfig.DATACEBO_BLUE,
+                'KeyUniqueness': PlotConfig.DATACEBO_GREEN
+
+            },
+            pattern_shape='Metric',
+            pattern_shape_sequence=['', '/'],
```

Comment:
Should we add a 3rd pattern shape to further distinguish the 3rd metric in the visualizations? @npatki